lib/faraday/log_filter/middleware.rb

```diff
@@ -0,0 +1,61 @@
+require "faraday"
+require 'forwardable'
+
+module Faraday
+  module LogFilter
+    class Middleware < Faraday::Response::Middleware
+      extend Forwardable
+
+      DEFAULT_OPTIONS = { headers: true, bodies: false }
+
+      def initialize(app, logger = nil, options = {})
+        super(app)
+        @logger = logger || begin
+          require 'logger'
+          ::Logger.new(STDOUT)
+        end
+        @options = DEFAULT_OPTIONS.merge(Hash(options[:log_options]))
+        @filter  = Array(options[:filter])
+      end
+
+      def_delegators :@logger, :debug, :info, :warn, :error, :fatal
+
+      def call(env)
+        env = Env.new(env, @filter)
+
+        info "#{env.http_verb.upcase} #{env.dump_url}"
+        debug('request') { env.headers } if log_headers?(:request)
+        debug('request') { env.dump_body } if env.has_body? && log_body?(:request)
+        super
+      end
+
+      def on_complete(env)
+        env = Env.new(env, @filter)
+
+        info('Status') { env.status }
+        debug('response') { env.headers } if log_headers?(:response)
+        debug('response') { env.dump_body } if env.has_body? && log_body?(:response)
+      end
+
+      private
+
+      def log_headers?(type)
+        case @options[:headers]
+        when Hash then @options[:headers][type]
+        else @options[:headers]
+        end
```

Comment:
is the case statement necessary when there's only a single branch?  What do you think of using something more succinct, like:

```ruby
def log_headers?(type)
  return @options[:headers][type] if @options[:headers].is_a?(Hash)
  @options[:headers]
end
```